contactql/parser_test.go

```diff
@@ -175,16 +227,11 @@ func TestEvaluateQuery(t *testing.T) {
 		// location field condition
 		{`state = kigali`, true},
 		{`state = "kigali"`, true},
-		{`state = "NY"`, false},
-		{`state ~ KIG`, true},
```

Comment:
we don't support `~` for fields.. but is there maybe some ES indexing for them? Am confused by https://github.com/nyaruka/rp-indexer/blob/master/indexer.go#L424

Reply:
Ooh, good catch. So this is basically indexing only the last part of something that looks like `Rwanda > Kigali > Kimihurura` .. so that you can query just `stat = Kimihurura`.

Since fields always contain the full path `Rwanda > Kigali > Kimihurura` then that's needed to allow that kind of querying. To tokenizer does not tokenize the remaining words though, so "Dakota" will not match "South Dakota" and "North Dakota".

So this is still a `=` but only on the leaf of the location.

Reply:
From rp-indexer.. 

```
"tokenizer": {
  "location_tokenizer": {
    "type": "pattern",
    "pattern": "(.* > )?([^>]+)",
    "group": 2
  },
  "trigram": {
    "type" : "ngram",
    "min_gram" : 3,
    "max_gram" : 3
  }
}
```

Ok to clarify..

1. The trigram stuff isn't being used? We can drop that in rp-indexer?
2. If a contact has `state` value of `Rwanda > Kigali > South Kimihurura` it would tokenize to `South Kimihurura` as a _single_ token, which would match if the user searched for exactly `state = "South Kimihurura"`

Shouldn't the tokenization go both ways? Looks like you can't search for something like `state = "Rwanda > Kigali > South Kimihurura"`.. or maybe my location data isn't right..

Reply:
1. Trigram tokenizer is being used for URNs, tabs are messed up in that file so it is hard to tell but that tokenizer is used for the trigram analyzer.
2. Correct.

So I was actually wrong, but wrong twice, so correct! I love it when that happens.

Turns out when we query, we query against the `_keyword` version of the field (both in Python and Golang):
    https://github.com/nyaruka/mailroom/blob/master/search/search.go#L224

This is not tokenized in any way, just like text fields it is just the full thing. So you would expect that to contain `Rwanda > Kigali > Kimihurura`, except that the indexer itself takes care of the trimming to just the appropriate level:
  https://github.com/nyaruka/rp-indexer/blob/master/indexer.go#L363

So what is indexed is indeed just `Kimihurura` and it is an exact term query there.

So no there's no way to type in the whole thing. I think we decided that if you had multiple wards with the same name it was more intuitive to do "state = kigali and district = kimihurura" then have people try to figure out the whole `Rwanda > Kigali > Kimihurura` nonesense.

Reply:
Ah ok and it seems `Contact.QueryProperty` already returns `Kimihurura` for `Rwanda > Kigali > Kimihurura` so looks like this all works fine already